digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -1425,11 +1425,24 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
       }.toSeq
 
       val recalculatedTerminated = ProjectSectionCalculator.assignTerminatedMValues(terminated, recalculated)
+      val assignedTerminatedRoadwayNumbers = assignTerminatedRoadwayNumbers(others ++ recalculatedTerminated)
       val originalAddresses = roadAddressService.getRoadAddressesByRoadwayIds((recalculated ++ recalculatedTerminated).map(_.roadwayId))
-      projectLinkDAO.updateProjectLinks(recalculated ++ recalculatedTerminated, userName, originalAddresses)
+      projectLinkDAO.updateProjectLinks(recalculated ++ assignedTerminatedRoadwayNumbers, userName, originalAddresses)
     }
   }
 
+  private def assignTerminatedRoadwayNumbers(seq: Seq[ProjectLink]): Seq[ProjectLink] = {
```

Comment:
Why terminated roadways are assigned a new roadway number here?
https://extranet.vayla.fi/wiki/display/VIITE/Roadway+Number#RoadwayNumber-TerminatingtheRoadwaywithHistory

Reply:
terminated links are being ignored and not sent to assignMValues method where the recalculation takes place. 
And we do assign roadway numbers during recalculation because its where we also adjust 2 tracks to match. But maybe in a near future we can think of spliting the logic of MValues and roadwaynumbers assignment since they are independent.